crates/livesplit-auto-splitting/src/process.rs

```diff
@@ -0,0 +1,67 @@
+use proc_maps::Pid;
+use read_process_memory::{CopyAddress, ProcessHandle};
+use sysinfo::{self, ProcessExt, System, SystemExt};
+
+#[derive(Debug)]
+pub enum Error {
+    // TODO: Doc Comments
+    ProcessDoesntExist,
+    ListModules,
+    ProcessOpening,
+    ModuleDoesntExist,
+    ReadMemory,
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+pub type Address = u64;
+
+pub struct Process {
+    handle: ProcessHandle,
+    pid: Pid,
+}
+
+impl std::fmt::Debug for Process {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Process").field("pid", &self.pid).finish()
+    }
+}
+
+impl Process {
+    pub fn with_name(name: &str, sysinfo: &mut System) -> Result<Self> {
+        sysinfo.refresh_processes();
+        let processes = sysinfo.process_by_name(name);
+        let pid = processes
+            .first()
+            .ok_or(Error::ProcessDoesntExist)
+            .map(|&p| p.pid())? as Pid;
+        let handle = pid.try_into().map_err(|_| Error::ProcessOpening)?;
+        Ok(Process { handle, pid })
+    }
+
+    pub fn is_open(&self, sysinfo: &mut System) -> bool {
+        sysinfo.refresh_processes();
```

Comment:
I just randomly stumbled upon this: https://blog.guillaume-gomez.fr/articles/2021-12-14+sysinfo%3A+version+0.22+and+FreeBSD+support

Apparently you can refresh just specific processes now.

Reply:
I implemented this in the dependency update commit I believe.